ship_dataset.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Aug 20 15:05:17 2019
+
+@author: j-bd
+"""
+import numpy as np
+
+from mrcnn import utils
+
+
+class ShipDataset(utils.Dataset):
+    '''class to manage the data for the training process with mrcnn'''
+    def load_ship(self, dataset_dir, subset):
+        ''' Reads pandas dataframe, extracts the annotations, and iteratively
+        calls the internal 'add_class' and 'add_image' functions to build the
+        dataset.'''
+        self.add_class("ship", 1, "ship")
+
+        for image in subset.ImageId.unique():
+            img_masks = subset.loc[subset['ImageId'] == image,
+                                   'EncodedPixels'].tolist()
+            self.add_image("ship",
+                           image_id=image,
+                           path=dataset_dir+image,
+                           width=768,
+                           height=768,
+                           img_masks=img_masks)
+
+
+    def mask_creation(self, mask_encode, shape=(768, 768)):
+        '''Create a mask from airbus string information to a numpy array'''
+        img = np.zeros(shape[0]*shape[1], dtype=np.uint8)
+        if not isinstance(mask_encode, str):
+            return img.reshape(shape)
+
+        mask_split = mask_encode.split()
+        starts, lengths = [np.asarray(
```

Comment:
```py
        starts, lengths = [
            np.asarray(x, dtype=int)
            for x in (mask_split[0::2], mask_split[1::2])
        ]
```
Is a good way to indent python comprehensions

Reply:
But anyway, could you maybe split `those` masks into 2 separate 1d ndarrays, instead of using that `[0::2]`/`[1::2]` trick everywhere? I thing it would simplify a lot your code everywhere.